homeassistant/components/lametric/notify.py

```diff
@@ -35,28 +34,41 @@ async def async_get_service(
     coordinator: LaMetricDataUpdateCoordinator = hass.data[DOMAIN][
         discovery_info["entry_id"]
     ]
-    return LaMetricNotificationService(coordinator.lametric)
+    return LaMetricNotificationService(coordinator)
 
 
 class LaMetricNotificationService(BaseNotificationService):
     """Implement the notification service for LaMetric."""
 
-    def __init__(self, lametric: LaMetricDevice) -> None:
+    def __init__(self, coordinator: LaMetricDataUpdateCoordinator) -> None:
         """Initialize the service."""
-        self.lametric = lametric
+        self.coordinator = coordinator
 
     async def async_send_message(self, message: str = "", **kwargs: Any) -> None:
         """Send a message to a LaMetric device."""
         if not (data := kwargs.get(ATTR_DATA)):
             data = {}
 
+        priority = NotificationPriority(data.get(CONF_PRIORITY, "info"))
+
+        if priority != NotificationPriority.CRITICAL:
+            # LaMetric devices will refuse non-critical messages when in screensaver
+            # mode, so if we have such a message, we get the latest coordinator data
+            # to see whether screensaver mode is enabled:
+            await self.coordinator.async_refresh()
```

Comment:
Asking for a refresh before each message is pretty inefficient, we should instead handle the error response when sending a message.